source/core/src/main/com/csse3200/game/components/CustomerSensorComponent.java

```diff
@@ -0,0 +1,121 @@
+     * @param sensorDistance The range of the sensor (how far it can detect)
+     */
+    public CustomerSensorComponent(short targetLayer, float sensorDistance) {
+        this.targetLayer = targetLayer;
+        this.sensorDistance = sensorDistance;
+    }
+
+    @Override
+    public void create() {
+
+        entity.getEvents().addListener("collisionStart", this::onCollisionStart);
+        entity.getEvents().addListener("collisionEnd", this::onCollisionEnd);
+        this.interactionComponent = entity.getComponent(InteractionComponent.class);
+
+        if (interactionComponent == null) {
+            
+        
+        super.create();
+    }
+  }
+
+    public void onCollisionStart(Fixture me, Fixture other) {
+        if (interactionComponent.getFixture() != me) {
+            return;
+        }
+        if (!isCustomer(other)) {
+            return;
+        }
+
+        if (isWithinDistance(other, sensorDistance)) {
+            collidingFixtures.add(other);
+        }
+        updateFixtures();
+    }
+
+    public void onCollisionEnd(Fixture me, Fixture other) {
+        if (interactionComponent.getFixture() != me) {
+            return;
+        }
+        if (!isCustomer(other)) {
+            return;
+        }
+
+        collidingFixtures.remove(other);
+        updateFixtures();
+    }
+
+    private void updateFixtures() {
+        Set<Fixture> toRemove = new HashSet<>();
+        for (Fixture fixture : collidingFixtures) {
+            float dist = getFixtureDistance(fixture);
+            if (dist > sensorDistance) {
+                toRemove.add(fixture);
+            } else if (closestDistance < 0 || dist < closestDistance) {
+                closestDistance = dist;
+                closestFixture = fixture;
+            }
+        }
+        collidingFixtures.removeAll(toRemove);
+
+        if (collidingFixtures.isEmpty()) {
+            closestFixture = null;
+            closestDistance = -1f;
+        }
+        if (closestFixture != null) {
+        } 
+        
+    }
+
+    private boolean isCustomer(Fixture fixture) {
+        Entity entity = ((BodyUserData) fixture.getBody().getUserData()).entity;
+        boolean isCustomer = entity.getComponent(CustomerComponent.class) != null;
+        return isCustomer;
+    }
+
+    private float getFixtureDistance(Fixture fixture) {
+        Vector2 sensorPosition = interactionComponent.getFixture().getBody().getPosition();
+        Vector2 fixturePosition = fixture.getBody().getPosition();
+        float distance = sensorPosition.dst(fixturePosition);
+        return distance;
+    }
+
+    public boolean isWithinDistance(Fixture fixture, float distance) {
+        float actualDistance = getFixtureDistance(fixture);
+        return actualDistance <= distance;
+    }
+
+    public Fixture getClosestCustomer() {
+        return closestFixture;
+    }
+    public InteractionComponent getInteractionComponent() {
+      return interactionComponent;
+  }
+}
```

Comment:
This could all just be inherited from the interaction component so that when you change how interaction works, it changes this as well. Everything except isCustomer and the function it is used in doesn't need to be here